### PR TITLE
Add GitHub Actions CI

### PR DESCRIPTION
**Pull request by wlynch**
Adds golang build/test and golangci-lint for basic CI coverage.

### CODE REVIEW

**Review by NatalieDoduc — APPROVED** [review]
LGTM, though I will admit it's not a well informed one. Will add someone who is perhaps more familiar with GitHub actions.

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    steps:
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v2
+        with:
+          go-version: ^1.13
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+
+      - name: Get dependencies
```
Comment (AlCutter) [inline .github/workflows/go.yml:20]: Do you need this bit? (dep is, um, *dep*recated :) and this repo uses modules to manage deps)

File: .github/workflows/golangci-lint.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+name: golangci-lint
+on:
```
Comment (AlCutter) [inline .github/workflows/golangci-lint.yml:2]: super-nit: vertical white space style is different between the two action config files in this PR, consider picking one style and adopting across both configs?

File: .github/workflows/go.yml (hunk 1 above)
Comment (wlynch) [reply .github/workflows/go.yml:16]: Removed.

File: .github/workflows/golangci-lint.yml (hunk 2 above)
Comment (wlynch) [reply .github/workflows/golangci-lint.yml:2]: Done.

**Review by wlynch — COMMENTED** [review]
Sorry about the delay! (have been in and out of the office for the last few weeks)

Everything should be resolved. Thanks!

**Review by AlCutter — APPROVED** [review]
LGTM!

**Pull request merged**